Store.js

```diff
@@ -140,14 +151,168 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
+					}
+				}
+			};
+		}),
+
+		/**
+		 * Discard change records for all listeners registered via `observe()`.
+		 */
+		discardChanges: dcl.superCall(function (sup) {
+			return function () {
+				sup.call();
+				if (this._observeResults) {
+					this._beingDiscarded = true;
+					this._observeResults[0].deliver();
+					if (this._hasItemObservable === true) {
+						this.observeCallback && Observable.deliverChangeRecords(this.observeCallback);
+					}
+					this._beingDiscarded = false;
+					return this.store;
+				}
+			};
+		}),
+
+		/**
+		 * Called to verify if an item is observable
+		 * @param item
+		 * @private
+		 */
+		_isItemObservable: function (item) {
+			return (Observable.test(item) || has("object-observe-api"));
+		},
+
+		/**
+		 * Called to verify if the array is observable
+		 * @param item
+		 * @private
+		 */
+		_isArrayObservable: function (item) {
+			return (ObservableArray.test(item) || has("object-observe-api"));
+		},
+
+		/**
+		 * Called to verify if the item respect the query conditions
+		 * @param item
+		 * @private
+		 */
+		_isQueried: function (item) {
+			for (var prop in this.query) {
+				if (item[prop] !== this.query[prop]) {
+					return false;
+				}
+			}
+			return true;
+		},
+
+		/**
+		 * Called when a modification is done on the array or its items.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallback: function (changeRecords) {
+			if (!this._beingDiscarded) {
+				for (var i = 0; i < changeRecords.length; i++) {
+					// array modified
+					if (Array.isArray(changeRecords[i].object)) {
+						this.observeCallbackArray(changeRecords[i]);
+						// one item of the array modified
+					} else {
+						var found = false;
+						for (var j = i + 1; j < changeRecords.length; j++) {
+							if (changeRecords[j].object === changeRecords[i].object) {
+								found = true;
+								changeRecords[j].oldValue = changeRecords[i].oldValue;
+							}
+						}
+						if (!found) {
+							this.observeCallbackItems(changeRecords[i]);
+						}
+					}
+				}
+			}
+		},
+
+		/**
+		 * Called when a modification is done on the array.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallbackArray: function (change) {
+			if (change.type === "splice") {
+				var j;
+				for (j = 0; j < change.removed.length; j++) {
+					var evtRemoved = {previousIndex: change.index};
+					this._itemRemoved(evtRemoved);
+				}
+				for (j = 0; j < change.addedCount; j++) {
+					var evtAdded = {
+						index: change.index + j,
+						target: this.store[change.index + j]
+					};
+					if (this.renderItems !== null && this.renderItems !== undefined) {
+						evtAdded.index = evtAdded.index <= this.renderItems.length ?
+							evtAdded.index : this.renderItems.length;
+					}
+					// affect the callback to the observe function if the item is observable
+					Observable.observe(this.store[change.index + j], this.observeCallback.bind(this));
```

Comment:
Should provide a way to stop observation (e.g. when the item is removed from array, when the widget is destroyed) of added array items.